### ISSUE TITLE
Fastboot error in the oauth2 authenticator

### ISSUE THREAD

**Issue by bakerac4** [simplabs/ember-simple-auth#1049]
Im seeing an error using the fastboot branch. 

```
There was an error running your app in fastboot. More info about the error: 
 TypeError: Cannot read property 'ajax' of undefined
```

I go to the fastboot vendor dist file and it brings me to [this line](https://github.com/simplabs/ember-simple-auth/blob/fastboot/addon/authenticators/oauth2-password-grant.js#L308) 

Farther up the stack I can see this is coming from the _refreshAccessToken method. Im using refresh token functionality, so it seems that the node application is trying to refresh the token. Could we just add a check to [this line](https://github.com/simplabs/ember-simple-auth/blob/fastboot/addon/authenticators/oauth2-password-grant.js#L319) and make sure that we aren't running in Fastboot mode?

**Comment by bakerac4** [comment]
Im continuing to bump into this. This time I saw it on session restore. 

I refreshed the page running FastBoot, and it saw that expires_at was not empty and expires_at is less than now. So it tried to refresh the access token. That being said I think we need this to succeed or else you won't be logged in? Perhaps using ember-network here might be the solution?

**Comment by marcoow** [comment]
Yes, ember-network would be the way to go for that. While FastBoot support is mostly ready in the `fastboot` branch there are still a bunch of unsupported edge cases like this one.

PR would be appreciated!

**Comment by bakerac4** [comment]
Hey @marcoow , I started work [here](https://github.com/bakerac4/ember-simple-auth/tree/fastboot). Im having trouble with the tests, and was wondering if you could give me insight into how I should remove jQuery's ajax and replace it with fetch.

**Comment by marcoow** [comment]
HM, I think you could either use Pretender instead of stubbing/spying on `jQuery`'s `ajax` method or find a way to stub the `fetch` method. Using Pretender is probably the better option, also in terms of overall quality and stability of the tests.

**Comment by bakerac4** [comment]
Fixed by #1066
